src/common/components/datepicker/Datepicker.tsx

```diff
@@ -98,11 +92,52 @@ const Datepicker: React.FC<Props> = ({
     datepickerClicked.current = false;
   };
 
-  const closeCalendar = () => {
-    setIsCalendarOpen(false);
-    onBlur();
+  const onDocumentFocusin = () => {
+    if (!isComponentFocused()) {
+      ensureCalendarIsClosed();
+    }
   };
 
+  React.useEffect(() => {
+    document.addEventListener('click', onDocumentClick);
+    document.addEventListener('keydown', handleDocumentKeyDown);
+    document.addEventListener('focusin', onDocumentFocusin);
+
+    return () => {
+      document.removeEventListener('click', onDocumentClick);
+      document.removeEventListener('keydown', handleDocumentKeyDown);
+      document.removeEventListener('focusin', onDocumentFocusin);
+    };
+  });
+
+  const isComponentFocused = () => {
+    const active = document.activeElement;
+    const current = container.current;
+
+    if (current && active instanceof Node && current.contains(active)) {
```

Comment:
This could also be written
`if (active instanceof Node && container.current?.contains(active))`

Reply:
hmm would 
if (current?.contains(active))
be enough?

Reply:
I think that was copy paste from another component

Reply:
Yeah, seems to work. That's even better.